### PR TITLE
MM-44859 : Screen reader invite members search results announcement

### PR DESCRIPTION
**Pull request by AbhinavVihan**
#### Summary
Screen reader invite members search results announcement

#### Ticket Link
Fixes: https://github.com/mattermost/mattermost-server/issues/21259
JIRA: https://mattermost.atlassian.net/browse/MM-44859

#### Related Pull Requests
None

#### Screenshots
None

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: components/widgets/inputs/users_emails_input.tsx [hunk 1]
```diff
@@ -276,6 +276,12 @@ export default class UsersEmailsInput extends React.PureComponent<Props, State>
             if (this.props.onUsersLoad) {
                 this.props.onUsersLoad(options);
             }
+            const optionWithLabel = options.forEach((option) => {
+                option.label = option.username;
+            });
+            return new Promise((resolve) => {
+                resolve(optionWithLabel);
+            });
```
Comment (M-ZubairAhmed) [inline components/widgets/inputs/users_emails_input.tsx:284]: can you please add explanation here as to why we do this?

File: components/widgets/inputs/users_emails_input.tsx [hunk 2]
```diff
@@ -276,6 +276,12 @@ export default class UsersEmailsInput extends React.PureComponent<Props, State>
             if (this.props.onUsersLoad) {
                 this.props.onUsersLoad(options);
             }
+            const optionWithLabel = options.forEach((option) => {
```
Comment (m1lt0n) [inline components/widgets/inputs/users_emails_input.tsx:279]: I've tested the changes with a screen reader and I still get undefined for the focused option.
I've tried out setting the option labels before setting state and it seems to work properly.

File: components/widgets/inputs/users_emails_input.tsx [hunk 3]
```diff
@@ -276,6 +276,12 @@ export default class UsersEmailsInput extends React.PureComponent<Props, State>
...
+                option.label = option.username;
+            });
+            return new Promise((resolve) => {
```
Comment (m1lt0n) [inline components/widgets/inputs/users_emails_input.tsx:282]: Why is this change needed? Setting the labels with `options.forEach....` is adequate. Its return value is undefined, so assigning it to `optionWithLabel` and returning the promise doesn't do something more than just setting the labels before setting state.

**Review by m1lt0n — CHANGES_REQUESTED** [review]

File: components/widgets/inputs/users_emails_input.tsx (hunk 2 above)
Comment (neallred) [reply components/widgets/inputs/users_emails_input.tsx:279]: Ditto what the others said. Adding a return value isn't necessary. Only setting label for the values that are sent to `callback` is needed. e.g. this works for me locally:

change

**Review by m1lt0n — APPROVED** [review]
Confirmed that the change fixes the issue with search results announcement.

File: packages/types/src/users.ts [hunk 4]
```diff
@@ -55,6 +55,7 @@ export type UserProfile = {
     terms_of_service_create_at: number;
     remote_id?: string;
     status?: string;
+    label?: string;
```
Comment (neallred) [inline packages/types/src/users.ts:58]: I'd rather add an `AccessibleProfile` type alias to `components/widgets/inputs/users_emails_input.tsx` like
```tsx
type AccessibleProfile = UserProfile & {label: string}
```
than add to the type here. This property isn't applicable in other contexts, and doesn't represent what is fetched from the server. The packages types are for broadly applicable and reused definitions.

**Review by M-ZubairAhmed — APPROVED** [review]

**Comment by jgilliam17** [comment]
Thank you @AbhinavVihan 
Tested, looks good to merge.
[E2E report](https://automation-dashboard.vercel.app/cycles/c9edef63-931b-4383-85e2-bcd89ae8088f) - no PR related failures.

**Review by jgilliam17 — APPROVED** [review]
